cli/services.go

```diff
@@ -90,55 +90,103 @@ func newServicesDeployCommand(ctx *common.Context) *cli.Command {
+		Usage:     common.SvcUndeployCmdUsage,
+		ArgsUsage: common.SvcUndeployArgsUsage,
 		Action: func(c *cli.Context) error {
 			environmentName := c.Args().First()
-			if len(environmentName) == 0 {
-				cli.ShowCommandHelp(c, "undeploy")
-				return errors.New("environment must be provided")
+			if len(environmentName) == common.Zero {
+				cli.ShowCommandHelp(c, common.UndeployCmd)
+				return errors.New(common.NoEnvValidation)
 			}
-			serviceName := c.Args().Get(1)
+			serviceName := c.Args().Get(common.SvcUndeploySvcFlagIndex)
 			workflow := workflows.NewServiceUndeployer(ctx, serviceName, environmentName)
 			return workflow()
 		},
 	}
 
 	return cmd
 }
+
 func newServicesLogsCommand(ctx *common.Context) *cli.Command {
 	cmd := &cli.Command{
-		Name:  "logs",
-		Usage: "show service logs",
+		Name:  common.LogsCmd,
+		Usage: common.SvcLogUsage,
 		Flags: []cli.Flag{
 			cli.StringFlag{
-				Name:  "service, s",
-				Usage: "service name to view logs for",
+				Name:  common.ServiceFlag,
+				Usage: common.SvcLogServiceFlagUsage,
 			},
 			cli.BoolFlag{
-				Name:  "follow, f",
-				Usage: "follow logs for latest changes",
+				Name:  common.FollowFlag,
+				Usage: common.FollowUsage,
 			},
 			cli.DurationFlag{
-				Name:  "search-duration, t",
-				Usage: "duration to go into the past for searching (e.g. 5m for 5 minutes)",
-				Value: 1 * time.Minute,
+				Name:  common.SearchDurationFlag,
+				Usage: common.SearchDurationUsage,
+				Value: common.DefaultLogDurationValue,
 			},
 		},
-		ArgsUsage: "<environment> [<filter>...]",
+		ArgsUsage: common.SvcLogArgUsage,
 		Action: func(c *cli.Context) error {
 			environmentName := c.Args().First()
-			if len(environmentName) == 0 {
-				cli.ShowCommandHelp(c, "logs")
-				return errors.New("environment must be provided")
+			if len(environmentName) == common.Zero {
+				cli.ShowCommandHelp(c, common.LogsCmd)
+				return errors.New(common.NoEnvValidation)
 			}
-			serviceName := c.String("service")
+			serviceName := c.String(common.SvcCmd)
 
-			workflow := workflows.NewServiceLogViewer(ctx, c.Duration("search-duration"), c.Bool("follow"), environmentName, serviceName, os.Stdout, strings.Join(c.Args().Tail(), " "))
+			workflow := workflows.NewServiceLogViewer(ctx, c.Duration(common.SearchDuration), c.Bool(common.Follow), environmentName, serviceName, os.Stdout, strings.Join(c.Args().Tail(), common.Space))
 			return workflow()
 		},
 	}
 
 	return cmd
 }
+
+func validateExecuteArguments(ctx *cli.Context) error {
+	environmentName := ctx.Args().First()
+	argLength := len(ctx.Args())
+
+	if argLength == common.Zero || len(strings.TrimSpace(environmentName)) == common.Zero {
+		cli.ShowCommandHelp(ctx, common.ExeCmd)
+		return errors.New(common.NoEnvValidation)
+	}
+	if argLength == common.ExeArgsSvcIndex {
+		cli.ShowCommandHelp(ctx, common.ExeCmd)
+		return errors.New(common.NoSvcValidation)
+	}
+	if len(strings.TrimSpace(ctx.Args().Get(common.ExeArgsSvcIndex))) == common.Zero {
```

Comment:
serviceName should be optional, passed as a `-s <service>` flag.  The default (if not provided) will be to pull the serviceName from the `mu.yml`, if file exists.